hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/AbstractLocalNode.java

```diff
@@ -62,9 +82,53 @@ public CompletableFuture<Void> mfFuture(@NonNull final MarkerFile markerFile) {
         return conditionFuture(() -> mfExists(markerFile), () -> MF_BACKOFF_MS);
     }
 
+    @Override
+    public Optional<Network> startupNetwork() {
+        return getStartupAddressBook(getExternalPath(DATA_CONFIG_DIR));
+    }
+
     protected abstract T self();
 
     private boolean mfExists(@NonNull final MarkerFile markerFile) {
         return Files.exists(getExternalPath(UPGRADE_ARTIFACTS_DIR).resolve(markerFile.fileName()));
     }
+
+    /**
+     * Tries to find any startup address book in the given directory or its {@code .archive} subdirectory.
+     * @param path the path to search
+     * @return the address book, if found
+     */
```

Comment:
😳 This sounds really scary to me. If something has been archived, there's probably a reason. Why is it necessary to retrieve anything from `.archive`?

Reply:
There is kind of similar [unarchiveGenesisNetworkJson](https://github.com/hashgraph/hedera-services/pull/16771/files#diff-27aeeed8d01c0c40855b43cc8540a9336c613c10c62b7fbc4a9ec0b13611695dR408) method in `StateChangesValidator` and there the validator creates a separate `Hedera` instance which will try to look up information from the archived genesis file (the HAPI suite archived it when started), so we need to recover it.

I think this one here is similar but I don't exactly remember which was the failing suite for it.